src/azure/cli/_profile.py

```diff
@@ -334,9 +331,7 @@ def persist_cached_creds(self):
 
             all_creds.extend(self._service_principal_creds)
             cred_file.write(json.dumps(all_creds))
-
-        if new_file:
-            os.chmod(self._token_file, 0o600)
+            cred_file.truncate()#to avoid reopening and write
```

Comment:
Using the flag appears do the same, but more consistent, so i take your suggestion. 
Note, barring the effect of umask, the whole change will always update the file mode, even a file which a user has specifically tweaked the permission. i thought it would be an issue, but i don't have strong opinions